op-service/sources/receipts_rcp.go

```diff
@@ -70,11 +70,11 @@ func NewRPCReceiptsFetcher(client rpcClient, log log.Logger, config RPCReceiptsC
 	}
 }
 
-func (f *RPCReceiptsFetcher) FetchReceipts(ctx context.Context, block eth.BlockID, txHashes []common.Hash) (result types.Receipts, err error) {
+func (f *RPCReceiptsFetcher) FetchReceipts(ctx context.Context, block eth.BlockID, txHashes []common.Hash, bInfo eth.BlockInfo) (result types.Receipts, err error) {
 	m := f.PickReceiptsMethod(len(txHashes))
 	switch m {
 	case EthGetTransactionReceiptBatch:
-		result, err = f.basic.FetchReceipts(ctx, block, txHashes)
+		result, err = f.basic.FetchReceipts(ctx, block, txHashes, bInfo)
 	case AlchemyGetTransactionReceipts:
 		var tmp receiptsWrapper
 		err = f.client.CallContext(ctx, &tmp, "alchemy_getTransactionReceipts", blockHashParameter{BlockHash: block.Hash})
```

Comment:
Also need validation on line 106

Reply:
imo we should _only_ validate on line 106 and remove the validation in the basic fetcher. The basic fetcher is not used standalone right now.

We could also give the basic fetcher a `validate bool`, only validate if this is true, set this in the ctor and then the `RPCReceiptsFetcher` can pass `false` when it constructs its basic fetcher. This way, if in theory in the future anyone ever built a standalone basic fetcher, they can choose whether they want it to validate or not. But not sure how important this is.

Reply:
I would like the basic fetcher to have the option to validate because it has the same shape as a receipt fetcher, but it does make sense to only validate once.